---
layout: page
category: cook
title: "IgBLAST examples"
order: 2
---

### Searching germline gene database. 

This example assumes that your current working directory has the following file structure:

```
igblastn
mouse_gl_V.nhr
mouse_gl_V.nin
mouse_gl_V.nog
mouse_gl_V.nsd
mouse_gl_V.nsi
mouse_gl_V.nsq
mouse_gl_D.nhr
mouse_gl_D.nin
mouse_gl_D.nog
mouse_gl_D.nsd
mouse_gl_D.nsi
mouse_gl_D.nsq
mouse_gl_J.nhr
mouse_gl_J.nin
mouse_gl_J.nog
mouse_gl_J.nsd
mouse_gl_J.nsi
mouse_gl_J.nsq
myseq
internal_data
optional_file 
```

To query one or more mouse Ig sequences (contained in myseq) against NCBI mouse germline gene 
database with standard text alignment result format, you can issue the following command:

```
./igblastn -germline_db_V mouse_gl_V -germline_db_J mouse_gl_J -germline_db_D mouse_gl_D -
organism mouse -domain_system kabat -query myseq -auxiliary_data optional_file/mouse_gl.aux -
show_translation -outfmt 3
```

you can type 
```
./igblastn -help
```
to see details on all input parameters and the default setting, 
particularly those under Ig-BLAST options.

### To query one or more mouse T cell receptor sequences (contained in myseq) against imgt mouse 
germline gene database:
```
./igblastn -germline_db_V imgt_tcr_db_v -germline_db_J imgt_tcr_db_j -germline_db_D imgt_tcr_db -
organism mouse -domain_system imgt -query myseq -ig_seqtype TCR -auxiliary_data 
optional_file/mouse_gl.aux -show_translation -outfmt 3
```

### To query one or human Ig sequences (contained in myseq) against custom database such as Andew 
Collins IGH repertoire database:
```
./igblastn -germline_db_V UNSWIgVRepertoire_fasta.txt -germline_db_J UNSWIgJRepertoire_fasta.txt -
germline_db_D UNSWIgDRepertoire_fasta.txt -organism human -domain_system kabat -query myseq  -
auxiliary_data optional_file/human_gl.aux -show_translation
```
Please be aware that you need to specify your own coding frame start and chain_type information (see 
details on optional files above) if your custom database has different sequence entries from NCBI or 
IMGT database.  

### Searching other databases in addition to germline database.
Igblast allows you to search an additional database (such as NCBI nr database) as well as the germline 
database at the same time.  You'll get hits from germline sequences followed by hits from non-germline 
database.
You MUST use -db option to specify the non-germline database which may not contain any sequences 
identifiers that exist in germline databases.  Note this option is ONLY for non-germline database 
(germline databases MUST be used with -germline_db_V, -germline_db_D or -germline_db_J option).  
To query one or more mouse Ig sequences (contained in myseq) against NCBI  nr database:
```
./igblastn -germline_db_V mouse_gl_V -germline_db_J mouse_gl_J -germline_db_D mouse_gl_D -
organism mouse -domain_system kabat -query myseq -auxiliary_data optional_file/mouse_gl.aux -
show_translation -outfmt 3 -db nr -remote 
```

Note the -remote option used with this search...-remote option directs igblast to send nr database 
searching to NCBI server which typically is much faster.
 
igblastp:
The parameters are similar to those of igblastn except it does not need germline D database, germline J 
database and optional file.  igblastp only performs search against V gene database.

Some examples:
  
### Searching mouse Ig sequence against mouse germline gene database
 
 ```
./igblastp -germline_db_V mouse_gl_V -query myseq.prot -outfmt 3 -organism mouse
```

### Searching other databases in addition to germline database.

```
./igblastp -germline_db_V mouse_gl_V -query myseq.prot -outfmt 3 -organism mouse -db nr â€“remote
```

Other notes:

The V gene domain annotation only works for IMGT annotation system for TCR sequences.


