---
layout: page
category: cook
title: "IgBLAST examples"
order: 2
---

There are two IgBlast command line programs, igblastn and igblastp.  The former is for nucleotide sequences and the latter is for protein sequences.

#### igblastn examples

##### Searching germline gene database 

These examples assume that your current working directory has the following file structure after installation:

```
bin
myseq
database
internal_data
optional_file 
```
Assuming you have put your germline gene blast database files under the directory "database",  to query a mouse sequence against NCBI mouse germline gene database with standard text alignment result format, you can issue the following command:

```
bin/igblastn -germline_db_V database/mouse_gl_V -germline_db_J database/mouse_gl_J -germline_db_D database/mouse_gl_D -organism mouse -query myseq -auxiliary_data optional_file/mouse_gl.aux -show_translation -outfmt 3
```

To add C gene identification function, simply add the following parameter to command line (assuming you have already downloaded ncbi_human_c_genes database from NCBI IgBlast FTP site):
```
-c_region_db database/ncbi_human_c_genes
```

To query a human sequence against IMGT human germline database:
```
bin/igblastn -germline_db_V database/imgt.Homo_sapiens.V.f.orf -germline_db_J database/imgt.Homo_sapiens.J.f.orf -germline_db_D database/imgt.Homo_sapiens.D.f.orf -organism human -query myseq -auxiliary_data optional_file/human_gl.aux -show_translation
``` 

To query a human sequence against custom database (using IGH repertoire from Andew Collins et al as an example)
```
bin/igblastn -germline_db_V database/UNSWIgVRepertoire_fasta.txt -germline_db_J database/UNSWIgJRepertoire_fasta.txt -germline_db_D database/UNSWIgDRepertoire_fasta.txt -organism human -query myseq  -auxiliary_data optional_file/human_gl.aux -show_translation
``` 

To query a mouse sequence against IMGT mouse TCR germline gene database:
```
bin/igblastn -germline_db_V database/imgt_tcr_db_v -germline_db_J database/imgt_tcr_db_j -germline_db_D database/imgt_tcr_db -organism mouse -query myseq -ig_seqtype TCR -auxiliary_data optional_file/mouse_gl.aux -show_translation -outfmt 3
```

You can type the following command to see details on all input parameters and the default setting, particularly those under Ig-BLAST options.
```
bin/igblastn -help
```


##### Searching other databases in addition to germline database.
Igblast allows you to search an additional database (such as NCBI nr database) as well as the germline 
database at the same time.  You'll get hits from the germline database followed by hits from the additional database.
Note that the additional database may not contain any sequences identifiers that also exist in germline databases.  

To query a mouse sequences against NCBI nr database, in addition to the mouse germline database:
```
bin/igblastn -germline_db_V database/mouse_gl_V -germline_db_J database/mouse_gl_J -germline_db_D database/mouse_gl_D -organism mouse -query myseq -auxiliary_data optional_file/mouse_gl.aux -show_translation -outfmt 3 -db nr -remote 
```

Note the -remote option used with this search...it directs igblast to send nr database 
searching to NCBI server which typically is much faster.
 
#### igblastp examples:
The parameters are similar to those of igblastn except it does not search germline D database or germline J 
database. The optional file is not needed.  
Some examples:
  
##### Searching a mouse sequence against mouse germline gene database
 
 ```
bin/igblastp -germline_db_V database/mouse_gl_V -query myseq.prot -outfmt 3 -organism mouse
```

##### Searching other databases in addition to germline database.

```
bin/igblastp -germline_db_V database/mouse_gl_V -query myseq.prot -outfmt 3 -organism mouse -db nr -remote
```



